File: Codes in C++/Array/Rotate Matrix by one element.cpp
#include <bits/stdc++.h>

void rotateMatrix(vector<vector<int>> &mat, int n, int m)
{   
        int depth=min(m,n);
        for(int i=0;i<depth/2;i++){
            int top=i,left=i,bottom=n-1-i,right=m-1-i;
            
                int temp=mat[top][left];

                for(int j=left+1;j<=right;j++)
                swap(mat[top][j],temp);
                for(int j=top+1;j<=bottom;j++)
                    swap(mat[j][right],temp);
                for(int j=right-1;j>=left;j--)
                    swap(mat[bottom][j],temp);
                for(int j=bottom-1;j>=top;j--)
                    swap(mat[j][left],temp);
        }

}

File: Codes in Python3/Array/Rotate Matrix by one element.py
def rotateMatrix(mat, n, m):
    depth = min(m, n)
    for i in range(depth // 2):
        top = i
        left = i
        bottom = n - 1 - i
        right = m - 1 - i

        temp = mat[top][left]

        for j in range(left + 1, right + 1):
            mat[top][j], temp = temp, mat[top][j]
        for j in range(top + 1, bottom + 1):
            mat[j][right], temp = temp, mat[j][right]
        for j in range(right - 1, left - 1, -1):
            mat[bottom][j], temp = temp, mat[bottom][j]
        for j in range(bottom - 1, top - 1, -1):
            mat[j][left], temp = temp, mat[j][left]
